src/sentry/static/sentry/app/views/ruleEditor/index.jsx

```diff
@@ -149,20 +151,40 @@ const RuleEditor = createReactClass({
     return !!error[field];
   },
 
+  handleEnvironmentChange(val) {
+    // If 'All Environments' is selected remove the environment property from the data
+    if (val === 'all') {
+      this.setState(state => {
+        const rule = {...state.rule};
+        delete rule.environment;
+        return {rule};
```

Comment:
what about:
```
let {environment, ...rule} = state.rule;
return {rule};
```

Reply:
I don't think that's better. It breaks our linting rules.